### DISCUSSION TITLE
Custom Terminal / Editor

### DISCUSSION THREAD

**Discussion by Mozart409** [johnlindquist/kit#173] [Ideas]
Hello,

I would like to open the source files in  [Terminus](https://github.com/Eugeny/terminus) and [Micro](https://github.com/zyedidia/micro). Is there a way to configure this? Thank you

**Comment by johnlindquist** [comment]
Oh sweet! I didn't know about those.

Terminal and iTerm both require AppleScript to launch and then write commands (examples below).

If you can find any AppleScript for launching Terminus and/or Micro, I'd be happy to add it to the project. Use "Script Editor" (an AppleScript editor that's on every mac) to test the scripts.

## Terminal
```osascript
tell application "Terminal"
  do script "echo hi"
  activate
end tell
```

## iTerm
```osascript
tell application "iTerm"
    if application "iTerm" is running then
        try
            tell the first window to create tab with default profile
        on error
            create window with default profile
        end try
    end if

    delay 0.1

    tell the first window to tell current session to write text "echo hi"
    activate
end tell
```

**Reply by Mozart409** [reply]
Nice. I will work on it 😄

**Reply by Mozart409** [reply]
1. Starting Terminus was easy

```sh
tell application "Terminus"
	activate
end tell
```

**Reply by johnlindquist** [reply]
Does `do script "echo hi"` part work? That's the tricky part

**Reply by Mozart409** [reply]
Yep here is an update

```sh
tell application "Terminus"
	activate
	
	set the clipboard to "echo Hello World!"
	tell application "System Events" to keystroke (the clipboard as text)
	tell application "System Events" to key code 76
	
end tell

```

Is this what you wanted? Ur able to write what ever you want and execute the command. I used key code 76 to also make it work for smaller apple keyboards.

**Reply by Mozart409** [reply]
Can we expect, that user has micro installed and already added to their path? So you can run micro <FILENAME> anywhere? If not then this would be way trickier

**Reply by johnlindquist** [reply]
1. I check to see which terminals are added to the path before prompting someone to select a default terminal
2. I avoid scripts that type the "Enter" key. Because it's async and errors do/will occur, you never know what you're really going to hit "Enter" on.

**Reply by Mozart409** [reply]
>     1. I check to see which terminals are added to the path before prompting someone to select a default terminal
> 
>     2. I avoid scripts that type the "Enter" key. Because it's async and errors do/will occur, you never know what you're really going to hit "Enter" on.

1. Ok so that part should be handled
2. Yeah, you are right. That is a problem. I will some more time to fix this. I will reply tomorrow...

Good night/day where ever you are

**Reply by Mozart409** [reply]
Its seems to be impossible to control terminus via the "do script" routine. Terminus does not have the functions build in to do that. I will focus on the texteditor micro now.

**Reply by johnlindquist** [reply]
I saw a comment with a mention of a "run cli". If there's any way to do `terminus "echo hi"`, then that would work too.
